tests/BuildScriptGenerator.Tests/Node/NodeBuildScriptGenerationTest.cs

```diff
@@ -452,6 +453,116 @@ public void GeneratedScript_DoesNotZipNodeModules_IfZipNodeModulesEnvironmentVar
+                packageInstallCommand: NpmInstallCommand,
+                runBuildCommand: "npm run build",
+                runBuildAzureCommand: "npm run build:azure",
+                hasProductionOnlyDependencies: true,
+                productionOnlyPackageInstallCommand: string.Format(
+                    NodeConstants.ProductionOnlyPackageInstallCommandTemplate,
+                    NpmInstallCommand),
+                compressedNodeModulesFileName: null,
+                compressNodeModulesCommand: null,
+                oryxAppInsightInjectCommand:null);
+
+            // Act
+            var snippet = scriptGenerator.GenerateBashBuildScriptSnippet(context);
+
+            // Assert
+            Assert.NotNull(snippet);
+            Assert.DoesNotContain("applicationinsights", snippet.BashBuildScriptSnippet);
+            Assert.Equal(
+                TemplateHelpers.Render(TemplateHelpers.TemplateResource.NodeBuildSnippet, expected),
+                snippet.BashBuildScriptSnippet);
+            Assert.True(scriptGenerator.IsCleanRepo(repo));
+        }
+
+        [Fact]
+        public void GeneratedScript_DoesNotConfigureAppInsight_IfNodeVersionCondition_Unsatisfied()
+        {
+            var nodeVersion = "6.9.3";
+            // Arrange
+            var scriptGenerator = GetNodePlatformInstance(defaultNodeVersion: nodeVersion);
+            var repo = new MemorySourceRepo();
+            repo.AddFile(PackageJsonWithBuildScript, NodeConstants.PackageJsonFileName);
+            Environment.SetEnvironmentVariable(Constants.AppInsightKey, "xyz");
+            var context = CreateScriptGeneratorContext(repo);
+            context.NodeVersion = nodeVersion;
+            var expected = new NodeBashBuildSnippetProperties(
+                packageInstallCommand: NpmInstallCommand,
+                runBuildCommand: "npm run build",
+                runBuildAzureCommand: "npm run build:azure",
+                hasProductionOnlyDependencies: true,
+                productionOnlyPackageInstallCommand: string.Format(
+                    NodeConstants.ProductionOnlyPackageInstallCommandTemplate,
+                    NpmInstallCommand),
+                compressedNodeModulesFileName: null,
+                compressNodeModulesCommand: null,
+                oryxAppInsightInjectCommand: null);
+
+            // Act
+            var snippet = scriptGenerator.GenerateBashBuildScriptSnippet(context);
+            Environment.SetEnvironmentVariable(Constants.AppInsightKey, string.Empty);
+
+            // Assert
+            Assert.NotNull(snippet);
+            Assert.DoesNotContain("applicationinsights", snippet.BashBuildScriptSnippet);
+            Assert.Equal(
+                TemplateHelpers.Render(TemplateHelpers.TemplateResource.NodeBuildSnippet, expected),
+                snippet.BashBuildScriptSnippet);
+            Assert.True(scriptGenerator.IsCleanRepo(repo));
+        }
+
+        [Theory]
+        [InlineData("8.0.0")]
```

Comment:
You can add more variations here (like 6.12.0)...is it expected to work for `6.12.1` as its a minor patch update?